dds/DCPS/RTPS/RtpsDiscovery.cpp

```diff
@@ -783,6 +800,19 @@ void RtpsDiscovery::request_remote_complete_type_objects(
   spdp->request_remote_complete_type_objects(remote_entity, remote_type_info, cond);
 }
 
+void RtpsDiscovery::write_stats(const MonotonicTimePoint&) const
+{
+  ACE_Guard<ACE_Thread_Mutex> guard(participants_lock_);
+  DCPS::InternalStatistics statistics;
+  for (DomainParticipantMap::const_iterator domain = participants_.begin(); domain != participants_.end(); ++domain) {
+    for (ParticipantMap::const_iterator part = domain->second.begin(); part != domain->second.end(); ++part) {
+      statistics.id = ("RtpsDiscovery " + DCPS::to_dds_string(domain->first) + ' ' + DCPS::to_string(part->first)).c_str();
```

Comment:
Do we want to allow the user to give a friendly name to a participant and use it (if set) here?

Reply:
For many applications, including RtpsRelay, the domain ID is sufficient.

I think it could be an option for future development.

Reply:
For posterity, when using security, it could be the subject name.

Reply:
Two participants could use the same security credentials